### ISSUE TITLE
mxCodec.decode won't add cells to graph

### ISSUE THREAD

**Issue by Lakshamana** [jgraph/mxgraph#331]
Hello! 
I'm trying to import cells in order to create a graph given a xml file. Below is the xml:
```
<mxGraphModel>
  <root>
    <mxCell id="2" value="Hello," parent="1" vertex="1">
      <mxGeometry x="20" y="20" width="80" height="30" as="geometry" />
    </mxCell>
    <mxCell id="3" value="World!" parent="1" vertex="1">
      <mxGeometry x="200" y="150" width="80" height="30" as="geometry" />
    </mxCell>
    <mxCell id="4" value="" parent="1" edge="1" source="2" target="3">
      <mxGeometry relative="1" as="geometry" />
    </mxCell>
  </root>
</mxGraphModel>
```
And my .js code (separated from the HTML):
```
const fs = require('fs')
import mxGraphFactory from 'mxgraph'

const {
  mxClient,
  mxGraph,
  mxUtils,
  mxCodec,
  mxEvent,
  mxRubberband
} = new mxGraphFactory()

function main(container) {
  if (!mxClient.isBrowserSupported()) { 
    mxUtils.error('Browser is not supported!', 200, false)
  } else {
    mxEvent.disableContextMenu(container)
    const graph = new mxGraph(container)
    new mxRubberband(graph)
    graph.getModel().beginUpdate()
    try {
      //deal with xml
      const xmlstr = fs.readFileSync('src/resources/diag.xml', 'utf-8')
      const doc = mxUtils.parseXml(xmlstr)      
      const codec = new mxCodec(doc)
      codec.decode(doc.documentElement, graph.getModel())
      console.log(graph.getModel())
    } finally {
      graph.getModel().endUpdate()
    }
  }
}

window.onload = main(document.getElementById('graphContainer'))
```
Currently using mxgraph in such a context:
mxgraph v3.9.12 (npm build) + parcel bundler
I've even created a function which uses graph.insertVertex/Edge to "manually" parse the xml and do the job (graph is created like a charm). However the point is to use mxCodec.decode.

**Comment by scoelli** [comment]
I had the same issue, I fixed it by using version 4.0.0 of mxgraph (`npm i jgraph/mxgraph#4.0.0`)

But now I have a issue that when I try to render vertex as childs of another vertex, the graph won't render them.

**Comment by davidjgraph** [comment]
I'll mark as fixed if 4.0.0 corrects correctly, that is the latest release.

**Comment by Lakshamana** [comment]
Thanks, @davidjgraph I will test this soon...

**Comment by Lakshamana** [comment]
Thanks, @davidjgraph I will test this soon...

> I had the same issue, I fixed it by using version 4.0.0 of mxgraph (`npm i jgraph/mxgraph#v4.0.0`)
> 
> But now I have an issue that when I try to render vertex as childs of another vertex, the graph won't render them.

true
